### ISSUE TITLE
Makefile doesn't link to pthread

### ISSUE THREAD

**Issue by ghost** [Infinidat/infinitrace#2]
```
gcc -L.  trace_dumper/trace_dumper.o trace_dumper/filesystem.o trace_dumper/events.o trace_dumper/sgio_util.o trace_dumper/internal_buffer.o trace_dumper/mm_writer.o trace_dumper/writer.o trace_dumper/write_prep.o trace_dumper/buffers.o trace_dumper/init.o trace_dumper/open_close.o trace_dumper/metadata.o trace_dumper/housekeeping.o trace_user_stubs.o -ltrace -ltraceutil -lparser -lsnappy  -lrt -o trace_dumper/trace_dumper 
trace_dumper/mm_writer.c:260: error: undefined reference to 'pthread_create'
trace_dumper/mm_writer.c:472: error: undefined reference to 'pthread_detach'
trace_dumper/mm_writer.c:468: error: undefined reference to 'pthread_join'
```

**Comment by ghost** [comment]
Added -pthread to linker options and it compiles fine:

``` make
trace_dumper: libtrace libtraceutil libsnappy libparser $(DUMPER_OBJS)
    gcc -L.  $(DUMPER_OBJS) -pthread -ltrace -ltraceutil -lparser -lsnappy $(EXTRA_LIBS) -o trace_dumper/trace_dumper 
```

**Comment by baruch** [comment]
This was fixed in commit dbb8ff801419c08ba3a2e44e3f9adebe51737a30
